src/main/java/com/homihq/db2rest/auth/ApiKeyAuthFilter.java

```diff
@@ -0,0 +1,51 @@
+package com.homihq.db2rest.auth;
+
+import com.homihq.db2rest.auth.service.ApiKeyVerifierService;
+import com.homihq.db2rest.auth.to.AuthInfo;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
```

Comment:
All these should be moved to a configuration class all beans must be created in that class
The configuration class is only active if `UNKEY_DEV_URL` parameter is present.